qiskit/compiler/transpile.py

```diff
@@ -121,6 +125,18 @@ def transpile(circuits: Union[QuantumCircuit, List[QuantumCircuit]],
             may not run.
         routing_method: Name of routing pass ('basic', 'lookahead', 'stochastic', 'sabre')
         translation_method: Name of translation pass ('unroller', 'translator', 'synthesis')
+        scheduling_method: Name of scheduling pass.
+            * ``'as_soon_as_possible'``: Schedule instructions greedily, as early as possible
+            on a qubit resource. alias: ``'asap'``)
+            * ``'as_late_as_possible'``: Schedule instructions late, i.e. keeping qubits
+            in the ground state when possible. (alias: ``'alap'``)
+            If ``None``, no scheduling will be done.
+        instruction_durations: Durations of instructions. Duration must be the number of ``dt``s.
+            Note that ``dt`` depends on backend configuration. Instruction is specified by
+            a pair of basic instruction name and its acting physical qubits.
+            E.g. [('cx', [0, 1], 1000), ('u3', [0], 300)]
+            Durations defined in ``backend.properties`` are used as default and
+            they are overwritten with the instruction_durations.
```

Comment:
Can we make some improvements to this:

1- It should be possible to "partially" specify gates, for example say `['u3', 300]` and have that match all qubits. If `['u3', 3, 500]` is given, then it will over-ride and make u3 duration on qubit 3 equal to 500.

2- Since delays can be expressed in s, ns, us, etc... why not these instruction durations too?

Reply:
1- We may need to update this docstring. As tested in [here](https://github.com/Qiskit/qiskit-terra/blob/f1a635a5d149df71db27d8637a73dc47990c97af/test/python/circuit/test_scheduled_circuit.py#L138), we can specify default duration say `('u3', None, 300)` and override it for a certain qubit (3) by `('u3', 3, 300)`.

2- Currently `instruction_durations` is designed to have "unitless integer durations" used for scheduling.

Reply:
Ok `('u3', None, 300)` sounds good. Please make sure the docstring is updated.